cmd/gateway-s3.go

```diff
@@ -102,12 +161,13 @@ type s3Objects struct {
 }
 
 // newS3Gateway returns s3 gatewaylayer
-func newS3Gateway(host string) (GatewayLayer, error) {
+func newS3Gateway(args cli.Args) (GatewayLayer, error) {
```

Comment:
if a gateway takes only one argument then we don't need to expand the scope here by adding more args. This refactor may have led to that.  But taking in input param as `host` is more meaningful here.